### PR TITLE
[Core] Fixed a potential buggy code where create_task is not invoked

### PR DESCRIPTION
**Pull request by rkooo567**
Signed-off-by: SangBin Cho <rkooo567@gmail.com>

## Why are these changes needed?

  When using create_task in a  "fire and forget" way, we should keep the references
  alive for the reliable execution. This API is used to fire and forget
  asynchronous execution.

Details: 
Screen Shot 2022-12-11 at 11 31 03 PM

This PR fixes the code that has the problem specified above ^. Also it implements a utility method to avoid such cases. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by simon-mo — APPROVED** [review]

**Review by architkulkarni — APPROVED** [review]
Did the end of the PR description get cut off accidentally? It ends with "Q: How should we"

File: python/ray/_private/utils.py [hunk 1]
```diff
@@ -1852,3 +1852,36 @@ def __exit__(self, exc_type, exc, exc_tb):
...
+    The above code doesn't guarantee to schedule the coroutine to the event loops
+
+    When using create_task in a  "fire and forget" way, we should keep the references
+    alive for the reliable execution. This API is used to fire and forget
+    asynchronous execution.
+
+    https://docs.python.org/3/library/asyncio-task.html#creating-tasks
+    """
+    task = get_or_create_event_loop().create_task(coroutine(*args, **kwargs))
+    global background_tasks
```
Comment (simon-mo) [inline python/ray/_private/utils.py:1879]: btw we don't this global here because we aren't setting the value here. Just calling methods.

**Comment by rkooo567** [comment]
@architkulkarni deleted that part. I don't remember what I was thinking to write lol

**Comment by rkooo567** [comment]
also added an unit test for run_backgroun_task

**Comment by rkooo567** [comment]
running a mac test now

**Pull request merged**